lib/cli/src/project_types.ts

```diff
@@ -112,7 +114,12 @@ export const supportedTemplates: TemplateConfiguration[] = [
   },
   {
     preset: ProjectType.VUE,
-    dependencies: ['vue', 'nuxt'],
+    // The Vue template only works with Vue or Nuxt under v3
+    // In a future update, a new Vue3 template will be added
+    dependencies: {
+      vue: (version) => lt(version, '3.0.0'),
```

Comment:
What's the end user behavior on a vue3 project? No project detected? Would be nice if we could detect that we don't support this version and explicitly warn why. Maybe out of scope for now.

Reply:
Interesting, I was letting it provide the "undetected" message, but maybe it makes sense to add an explicit message for Vue3. Do you think that's worthwhile, given the behavior will only ever exist in an alpha version?

Reply:
I was thinking more generally if there are things we know we don't support. I think for vue3 it's fine to add this then remove it, but then why add in the first place?

Reply:
Ah, yeah there could be a "framework" that is like `ProjectType.UNSUPPORTED` and it could have some specific dependencies it looks for and then provides a fairly generic message for unsupported frameworks, similar to how UNDETECTED works!

I will add that, great idea!

Reply:
@shilman I added this logic in a separate commit. Please let me know what you think!